### PR TITLE
Upgrade to core 1.75

### PR DESCRIPTION
**Pull request by Simon-Laux**
core commit: https://github.com/deltachat/deltachat-core-rust/commit/a1f04d2129441b6235ca6546e5152f30a3e311a4

no new tests for now, if you want to write them feel free to do so.

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -2462,6 +2470,14 @@ NAPI_METHOD(dcn_msg_set_quote) {
   NAPI_RETURN_UNDEFINED();
 }
 
+NAPI_METHOD(dcn_msg_remove_quote) {
```
Comment (r10s) [inline src/module.c:2473]: afail, most `dc_*` core-functions map 1:1 to `dcn_*` node functions.

unless there are good reasons to not do so, i'd stay with this approach and keep things consistent. so, remove `dcn_msg_remove_quote()` can use `dcn_msg_set_quote(NULL)` instead. that would also remove some lines of code.

Comment (Simon-Laux) [reply src/module.c:2473]: I spend like 2 hours trying to do this properly, unless you find a way to detect JS's null object we will stay with this workaround.
Or we could sometime rewrite the whole nodebindings properly with napi.rs, but not now.

Comment (r10s) [reply src/module.c:2473]: > I spend like 2 hours trying to do this properly, unless you find a way to detect JS's null object in c we will stay with this workaround.

i'd say that just the same as with `dcn_set_draft` that also passes a null-message-pointer from js to c.

i always do these kind of things like that, copy+paste a similar function :) eg. also for the java-bindings where i also tend to forget the details :)

but sure, as said, no blocker. if the current doubled function works and is tested, it's fine. but maybe sth. for the future.

Comment (Simon-Laux) [reply src/module.c:2473]: thanks for pointing that out,
```c
dc_msg_t* dc_msg_quote = NULL;
napi_get_value_external(env, argv[1], (void**)&dc_msg_quote);
```
seems to work, thanks 👍

**Comment by r10s** [comment]
core 1.71 is finally released and tagged :)

**Comment by r10s** [comment]
meanwhile, there is core72, core71 has a bug wrt backup-import and should not be used.

File: src/module.c [hunk 2]
```diff
@@ -2764,7 +2837,58 @@ NAPI_METHOD(dcn_provider_get_status) {
...
+  napi_value jsbuffer;
+  NAPI_STATUS_THROWS(napi_create_buffer_copy(env,
+                                    size,
+                                    &data,
+                                    NULL,
+                                    &jsbuffer))
+
+  free(data);
+
+  return jsbuffer;
```
Comment (Simon-Laux) [inline src/module.c:2889]: buffer length is correct now, but it does not return the right data.
any ideas what my mistake is here? @link2xt @r10s @flub @Jikstra
(I mention people that I think could know the solution)

Comment (Simon-Laux) [reply src/module.c:2889]: ah I was just stupidly adding an `&` before `data`.

File: src/module.c [hunk 3]
```diff
@@ -2764,7 +2837,58 @@ NAPI_METHOD(dcn_provider_get_status) {
...
+
+  size_t size;
+  char* data = dc_msg_get_webxdc_blob(dc_msg, filename, &size);
+  free(filename);
+
+  // https://nodejs.org/api/n-api.html#napi_create_buffer_copy
+  napi_value jsbuffer;
+  NAPI_STATUS_THROWS(napi_create_buffer_copy(env,
+                                    size,
+                                    &data,
```
Comment (link2xt) [inline src/module.c]: Pass just the `data` here, not `&data`. `data` is already a pointer to data, and you pass a pointer to pointer here.

File: src/module.c [hunk 4]
```diff
@@ -2764,7 +2837,60 @@ NAPI_METHOD(dcn_provider_get_status) {
...
+  if (data == NULL) {
+    NAPI_STATUS_THROWS(napi_get_null(env, &jsbuffer));
+  } else {
+    // https://nodejs.org/api/n-api.html#napi_create_buffer_copy
+    NAPI_STATUS_THROWS(napi_create_buffer_copy(env,
+                                      size,
+                                      data,
+                                      NULL,
+                                      &jsbuffer))
+    free(data);
```
Comment (r10s) [inline src/module.c:2888]: that should be `dc_str_unref()` - this is basically only `free()`, however, it makes sure it is called with the same scope as `malloc()` (might be, malloc here and in core use different heaps, we had similar issues before, therefore, `dc_str_unref()` exists :)

**Comment by r10s** [comment]
no day without a new core, the number of the day is core75 :)

File: src/module.c [hunk 5]
```diff
@@ -1977,6 +2014,23 @@ NAPI_METHOD(dcn_lot_get_timestamp) {
...
 
+NAPI_METHOD(dcn_msg_get_parent) {
+  NAPI_ARGV(1);
+  NAPI_DC_MSG();
+
+  napi_value result;
+  dc_msg_t* msg = dc_msg_get_parent(dc_msg);
+
+  if (msg == NULL) {
+    NAPI_STATUS_THROWS(napi_get_null(env, &result));
```
Comment (Jikstra) [inline src/module.c:2025]: you really wanna throw here and not just return null/false/-1 whatever? We could always still throw in the higher bindings

**Pull request merged**